starfyre/parser.py

```diff
@@ -150,25 +158,49 @@ def handle_endtag(self, tag):
         # if the tag is not found in the generic tags and custom components
         if tag not in self.generic_tags and tag not in self.components:             
             raise UnknownTagError(f'Unknown tag: "{tag}". Please review line {self.lineno} in your "{self.component_name}" component in the pyml code.')
-
-        # we need to check if the tag is a default component or a custom component
-        # if it is a custom component, we get the element from the custom components dict          
-
-        if tag not in self.generic_tags and tag in self.components:
-            component = self.components[tag]
-            tag = component.tag
         
-        endtag_node = self.stack.pop()  
+        endtag_node = self.stack.pop()
         self.current_depth -= 1
+
         if endtag_node.tag != "style" and endtag_node.tag != "script":
+            # basically finding and replacing the slot with
+            # the actual content
+            new_children = []
+            is_slot_used = False
+            for child_component in endtag_node.children:
+                if child_component.is_slot_component:   #We check each child in the entag_node list for the slot components 
+                    new_children.extend(self.current_children)
+                    is_slot_used = True
+                else:
+                    new_children.append(child_component)
+
+            if not is_slot_used and len(self.current_children) > 0:
+                #If there arent slot tag on the template.fyre and
+                # the current_children is not empty, means that the user forget 
+                # to add the slot tag, so we add the content to the end and let the user knows
+                new_children.extend(self.current_children)
+                print("Appending at the end of the stack as slot position not specified.")
+
+            if endtag_node.original_name != endtag_node.tag:
+                # if the tag is not found in the generic tags but found in custom components
+                # we need to replace the tag with the actual component
+                # and add the children to the component
+                # @suelen can you come up with a better explanation for this?
```

Comment:
cc @SuelenKarbivnychyy

Reply:
I have made the logic cleaner. Can you please write an explanation here?

Reply:
> I have made the logic cleaner. Can you please write an explanation here?

@sansyrox 
My approach is that we should process slot elements when we are closing the tag for our custom component (`</parent>`), like in the code bellow

```python
        if endtag_node.tag != "style" and endtag_node.tag != "script":
            if endtag_node.original_name != endtag_node.tag: #this node is for "custom" component
... #slot processing
```
With current proposal of moving this check after processing slot elements, I'm not sure, I understand how is it working.

Reply:
@SuelenKarbivnychyy , did you go through the comments in the code?

> closing the tag for our custom component (</parent>), like in the code bellow

This approach eliminates the possibility of us having `slots` in non custom components.

Reply:
> @SuelenKarbivnychyy , did you go through the comments in the code?
> > closing the tag for our custom component (), like in the code bellow
> 
> This approach eliminates the possibility of us having `slots` in non custom components.

@sansyrox 
- Yes, processing slot component is correct/well explained on the comments and I understand it.

- But What do you mean by `having `slots` in non custom components.`?

- Related to your question `I have made the logic cleaner. Can you please write an explanation here?`

I can not explain your reasons/or why is better to have the check for `closing custom compoment tag` after processing slots component.

From my understanding, slot tag is meant to be specified on the custom component definition "`parent.fyre definition`" so we must check if the `closing tag is a custom tag` before processing slot.